src/misc/lv_style.h

```diff
@@ -135,6 +135,7 @@ enum _lv_grad_dir_t {
     LV_GRAD_DIR_NONE, /**< No gradient (the `grad_color` property is ignored)*/
     LV_GRAD_DIR_VER,  /**< Vertical (top to bottom) gradient*/
     LV_GRAD_DIR_HOR,  /**< Horizontal (left to right) gradient*/
+    LV_GRAD_DIR_DIAGONAL,  /**< Diagonal (any angle) gradient*/
```

Comment:
How can we configure the angle?

Reply:
The angle of linear gradient is defined by its area. if the area is {{0,0}, {100, 100}}, the its angle is 45-degre.

Reply:
How to describe e.g. `30°` for a 100x40 sized widget?

Reply:
I think I understand your question. Maybe we should add `angle` in `lv_grad_dsc_t` to describe the angle of the linear gradient. But `angle` only work when `grad_area` is {{0,0}, {0,0}}, because `grad_area` will determine the angle. What do you think about this?

Reply:
I don't understand how the area determines the angle. 

In CSS it's described like:
```
linear-gradient(30deg, red 10% , blue 80%)
```

Meaning  red until 10% of the length, gradient until 80%, and blue at the last 20%.

Reply:
I see now. So basically it's 2 points. What if we change `frac` to a point in lv_gradient_stop_t?

It could be either percentage or pixel.